package/MDAnalysis/core/universe.py

```diff
@@ -1700,6 +1778,103 @@ def guess_TopologyAttrs(
+            of the list should be equal to the number of atoms in the Universe.
+            If `None`, the original segids will be used.
+
+        Raises
+        ------
+        AssertionError
+            If the length of the provided atomwise_resids or atomwise_segids
+            does not match the number of atoms in the Universe.
+
+        Notes
+        -----
+        First, the function will check if resids or segids is provided.
+        If both resids and segids are not provided (`None`), it will do nothing.
+        If only one of them is provided, it will use the original values for the
+        other one. If both are provided, it will use the provided values for
+        both resids and segids.
+        The function will then update the topology by a new generated topology
+        with new values of the resids and segids.
+        Finally, the corresponding new `ResidueGroup` and `SegmentGroup` will be
+        created by the updated topology.
+
+        Examples
+        --------
+        To set custom segment IDs for the segments of the Universe::
+
+            atomwise_segids = ['A', 'A', 'B', 'B']
+            u.set_groups(atomwise_segids=atomwise_segids)
+
+            # Now the Universe has two segments with segIDs 'A' and 'B'
+            u.segments
+            >>> <SegmentGroup with 2 segments>
+
+        .. versionadded:: 2.10.0
+        """
+        if (atomwise_resids is None) and (atomwise_segids is None):
+            warnings.warn("Do nothing. Please provide atomwise_resids or "
+                          "atomwise_segids.")
+            return
+
+        else:
+            # resids
+            if atomwise_resids is None:
+                atomwise_resids = self.atoms.resids
+
+            else:
+                # check the length of atomwise_resids
+                assert len(atomwise_resids) == self.atoms.n_atoms, \
+                    "The length of atomwise_resids should be the same as " \
+                    "the number of atoms in the universe."
+
+                self.atomwise_resids_orig = self.atoms.resids
+                logger.info("The new resids replaces the current one. "
+                            "The original resids is stored in "
+                            "atomwise_resids_orig.")
+
+            # segids
+            if atomwise_segids is None:
+                atomwise_segids = self.atoms.segids
+
+            else:
+                # check the length of atomwise_segids
+                assert len(atomwise_segids) == self.atoms.n_atoms, \
+                    "The length of atomwise_segids should be the same as " \
+                    "the number of atoms in the universe."
+
+                self.atomwise_segids_orig = self.atoms.segids
+                logger.info("The new resids replaces the current one. "
+                            "The original segids is stored in "
+                            "atomwise_segids_orig.")
```

Comment:
you can remove that first `else:` line and de-indent the rest of the block, since you have a `return` at the end of the `if` block